hs628_tools/random-forest.py

```diff
@@ -0,0 +1,30 @@
+import numpy as np
+from sklearn.model_selection import train_test_split
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+
+def rf(features, target, model='classifier'):
+    train_features, train_target, test_features, test_target = train_test_split(features, target, test_size=0.25, random_state=13)
+    if model == 'classifier':
+        clf = RandomForestClassifier(n_estimators=10, criterion='gini', max_features='auto',
+                                     n_jobs=1, random_state=13, class_weight='balanced')
+        clf.fit(train_features, train_target)
+        clf.predict(test_features)
+        predictions = clf.predict(test_target)
+        scores = list(zip(train_features, clf.feature_importances_))
+
+        return predictions, scores
+    elif model == 'regressor':
+        reg = RandomForestRegressor(n_estimators=20, max_features='auto')
```

Comment:
Hi Douglas, I think you code is straightforward on its goal. I like your idea on allowing users to select classifier or regressor. I just wanted to add three points on your work.
- Even your code is clear to understand, like Nirvesh said, it would be helpful if you add documentations especially on why you picked/ended up particular parameters: e.g. why you picked "gini" for criterion of classifier. I guess you did as the supported criteria are “gini” for the Gini impurity and “entropy” for the information gain. But adding those comments/documents will help to understand your structure and reasoning on the code.
- Based on that, I`d like to suggest you add "mse" as the default criterion for 'regressor', like you set "gini" for classifier.
- I think your idea and coding would be improved much better if you add "Grid Search" function to optimize fitting. I`ll add url here for your reference.(https://www.fabienplisson.com/random-forest-and-grid-search/)